src/btree/bt_cursor.c

```diff
@@ -945,11 +1092,13 @@ err:	if (ret == WT_RESTART) {
 	 * To make this work, we add a field to the btree cursor to pass back a
 	 * pointer to the modify function's allocated update structure.
 	 */
-	if (ret == 0)
+done:	if (ret == 0)
 		WT_TRET(__wt_kv_return(session, cbt, cbt->modify_update));
-
-	if (ret != 0)
+	else {
 		WT_TRET(__cursor_reset(cbt));
+		WT_CURFILE_OP_POP;
+	}
```

Comment:
This code was previously structured to handle an error from `__wt_kv_return` the same as other errors (and reset the cursor).  Switching to an `else` has lost that, can it go back to an `if (ret != 0)`?